src/main/java/seedu/address/model/self/Weight.java

```diff
@@ -0,0 +1,61 @@
+package seedu.address.model.self;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents the user's weight in the diet tracker of EYLAH.
+ * Guarantees: immutable; is valid as declared in {@link #isValidWeight(String)}
+ */
+public class Weight {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Weight should only contain numbers, and it should not be blank. It is to be input in kilograms.";
+
+    /*
+     * The first character of the weight must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}]*";
+
+    public final String weight;
+    public final Float wt;
+
+    /**
+     * Constructs a {@code Weight}.
+     *
+     * @param weight A valid weight.
+     */
+    public Weight(String weight) {
+        requireNonNull(weight);
+        checkArgument(isValidWeight(weight), MESSAGE_CONSTRAINTS);
+        this.weight = weight.toString();
```

Comment:
Same as height, no need to call toString on the weight as it is already a String. Causing Codacy errors.